### PR TITLE
Logger instead of print in lr scheduler

### LINKED ISSUE #100847: logger instead of print in lr_scheduler.py
### 🚀 The feature, motivation and pitch

In lr_scheduler.py, there are a couple calls to print().  For example, `ReduceLROnPlateau._reduce_lr` calls `print`.  If it called a logger instead, the logs could be turned off, or at least would show up in python logging streams as part of stderr.

### Alternatives

_No response_

### Additional context

_No response_

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

### PR DESCRIPTION
**Pull request by gerhean**
Fixes #100847

Use `logger.warning()` instead of `logger.info()` as `logger.warning()` prints to stdout by default, which best preserves the previous behavior. 

I'm not sure if the docstring change is appropriate as it might be confusing to people who are not familiar with python logging.

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Review by janeyx99 — COMMENTED** [review]
Thanks @gerhean for putting up a PR. However, I'd like to redirect to some relevant followup discussion on the issue before we take further steps here.

**Review by Skylion007 — CHANGES_REQUESTED** [review]
Waiting on feedback from @janeyx99
